docs/component-testing.md

```diff
@@ -0,0 +1,138 @@
+
+### Implementation options for component testing
+
+Component testing is a type of testing in which the logic of a web component is tested in isolation from the web page in which it is used. In order to perform such a test, you need to be able to render the component correctly. [JSDom](https://github.com/jsdom/jsdom) is often used for this task (also used inside Jest), which renders web components using the Node.js virtual renderer, i.e. without using a real browser. On the one hand, it works faster (the browser does not launch), and on the other hand, it is less stable, since the checks are not performed in a real browser. The second popular solution is to use a very fast dev server [Vite](https://vitejs.dev/), which supports many frameworks (React, Vue, Svelte, ...) and is responsible for rendering components in isolation.
+
+We chose the option using Vite, as this approach ensures that the page is tested more closely to reality (as if the user had opened it). At the same time, the tests themselves run a little longer than in the JSDom. But the most important thing for us is the stability and reproducibility of test results, so the choice was obvious.
+
+### How to use it?
+
+We will set up testing of react components written in Typescript. Therefore, first of all, we will install the necessary dependencies:
+
+```bash
+npm i testplane vite @vitejs/plugin-react @testing-library/react --save-dev
+npm i react --save
+```
+
+Now let's create a Vite config in which we will connect the plugin to support React. Example:
+
+```typescript
+// vite.config.ts
+import { defineConfig } from 'vite';
+import react from '@vitejs/plugin-react';
+
+export default defineConfig({
+    plugins: [
+        react(),
+    ]
+});
+```
+
+After that, we will configure the tests to run in the browser. To do this, specify the [testRunEnv](./config.md#testrunenv) option. Example:
+
+```typescript
+// .testplane.conf.ts
+export const {
+    // ...
+    system: {
+        // ...
+        testRunEnv: ['browser', { viteConfig: './vite.config.ts' }],
+    },
+    sets: {
+        linux: {
+            files: [
+                'src/tests/**/*.testplane.tsx'
+            ],
+            browsers: [
+                'chrome'
+            ]
+        },
+    },
+}
+```
+
+And in the end, we can write a test in which we simply output the `document` value to the console without using the `browser.execute` command:
```

Comment:
> in which

"where"

> browser.execute

Maybe we should make it hyperlink to wdio command? Not everyone knows what it does